### PR TITLE
Conductor: fix config loading from environment to have more control

### PR DESCRIPTION
**Pull request by steezeburger**
## Summary
This PR refactors how we load values from the environment so that we have more control on which envars are used.
We can now define the `envar_prefix` we want as well as a `test_envar_prefix` so that we have full control over which envar names should be part of the config.

## Background
A user's locally loaded environment should not affect test outcomes.

## Changes
- added `envar_prefix` argument to `from_environment`
- added `test_envar_prefix` to `populate_environment_from_example`

## Testing
- I fixed the test that was erroneously failing previously.

## Breaking Changelist
- You must now pass in an `envar_prefix` to `Config::from_environment`

closes https://github.com/astriaorg/astria/issues/386

### CODE REVIEW

**Review by sambukowski — APPROVED** [review]

File: crates/astria-conductor/src/config.rs [hunk 1]
```diff
@@ -66,10 +66,14 @@ where
 }
 
 impl Config {
-    fn from_environment() -> Result<Config, figment::Error> {
+    fn from_environment(prefix: Option<&str>) -> Result<Config, figment::Error> {
```
Comment (noot) [inline crates/astria-conductor/src/config.rs:69]: it appears `None` is the only value passed in here (excluding tests), maybe just remove the param and the `Some` case?

**Review by SuperFluffy — CHANGES_REQUESTED** [review]
IMO the code can be simplified by not using the option but instead changing `Config::from_environment` to take a `&str`, and then defining a hardcoded string to use in `fn get`.

Also, I think it would be great if we expanded this to the other crates.

**Review by joroshiba — APPROVED** [review]
Generally LGTM, although I agree with comments that we should set a const and always take a string instead of using `None` value.

Assuming that is taken care of, approved.

**Comment by steezeburger** [comment]
> IMO the code can be simplified by not using the option but instead changing `Config::from_environment` to take a `&str`, and then defining a hardcoded string to use in `fn get`.
> 
> Also, I think it would be great if we expanded this to the other crates.

Totally! I made an issue for pulling this out into its own crate https://github.com/astriaorg/astria/issues/398

**Review by SuperFluffy — APPROVED** [review]
Looking good, ty for the changes

**Pull request merged**